go/vt/sqlparser/sql.y

```diff
@@ -647,19 +696,27 @@ common_table_expression:
 union_lhs:
   base_select
   {
+    if $1.HasIntoDefined() {
+      yylex.Error(fmt.Errorf("INTO clause is not allowed").Error())
+      return 1
+    }
     $$ = $1
   }
-| openb select_statement closeb
+| openb select_statement_with_no_into closeb
   {
     $$ = &ParenSelect{Select: $2}
   }
 
 union_rhs:
   base_select_no_cte
   {
+    if $1.HasIntoDefined() {
```

Comment:
`into` is still allowed in the right-hand side of a union statement, but not the left hand side. This reads like it's preventing it on the right-hand side, too? Let me know if I'm just missing something. It seems like your tests correctly covered this case with `into` on the right-hand side, just wanted to check that I wasn't missing something here.

Reply:
For `union` statement, `into_opt` should be wrapping around the whole union statement. MySQL allows only the last union (rightmost union) to have `into_opt`. There is a conflict on where right hand side of union that has the `into_opt` can be inside of another union, which is not allowed. Although MySQL allows into option before order by or limit opts that comes after `base_select` rule, it is deprecated and will be removed in future, so we removed supporting the middle `into_opt`. That leaves us to have `into_opt` only at the end of the statement specifically for `union` statements, but have before `from_opt` and at the end of query for `select` statements.